driver/src/main/java/org/neo4j/driver/Session.java

```diff
@@ -47,6 +47,13 @@
      */
     Transaction newTransaction();
 
+    /**
+     * Detect whether the session is currently open or not.
+     *
+     * @return true if the session is open, false if it has been {@link #close() closed}.
+     */
+    boolean isOpen();
```

Comment:
Interesting, when googling `fairynuff`, I got "Fairy Nuff is a fairy who studies medicine in Zanaris, found in her house north of the bank. She...", then I realized it is fair enough :)